src/indextypes.jl

```diff
@@ -95,7 +95,23 @@ end
 """
     delete(model::ModelLike, index::Index)
 
-Delete the referenced object from the model.
+Delete the referenced object from the model. Throw [`DeleteNotAllowed`](@ref) if
+if `index` cannot be deleted.
+
+The following modifications also take effect if `Index` is [`VariableIndex`](@ref):
+* If `index` used in the objective function, it is removed from the function;
+  see [`Utilities.remove_variable`](@ref).
```

Comment:
It's strange to refer to Utilities in the definition of an MOI function (except as an example of an implementation). Could we put the full definition here (and perhaps let `Utilities.remove_variable` reference it)?

Reply:
In fact, it is not explained better at `Utilities.remove_variable` so I just replaced it with a def